### ISSUE TITLE
batchnorm on chunks causes cpu memory leak (but not gpu)?

### ISSUE THREAD

**Issue by hengck23** [pytorch/pytorch#125649]
### 🐛 Describe the bug
I am training in cuda with large batchsize=5000 and observed that my cpu memory is increasing when monitored via htop. Gpu memory is stable with nvidia-smi.

Since my simple network consists of only linear, conv1d and batchnorm1d layers, I can find the root cause by disabling them one by one. It turns out that the cause of memory leak is batchnorm1d.

Why? is it a bug?

### Versions

PyTorch version: 2.0.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 8.4.0-1ubuntu1~18.04) 8.4.0
Clang version: Could not collect
CMake version: version 3.26.4
Libc version: glibc-2.27

Python version: 3.10.9 (main, Mar  1 2023, 18:23:06) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.4.0-47-generic-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: 11.7.64
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: Quadro RTX 8000
GPU 1: Quadro RTX 8000

Nvidia driver version: 525.125.06
cuDNN version: Probably one of the following:
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn.so.8.9.3
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn_adv_infer.so.8.9.3
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn_adv_train.so.8.9.3
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn_cnn_infer.so.8.9.3
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn_cnn_train.so.8.9.3
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn_ops_infer.so.8.9.3
/usr/local/cudnn-linux-x86_64-8.9.3.28_cuda11-archive/lib/libcudnn_ops_train.so.8.9.3
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              72
On-line CPU(s) list: 0-71
Thread(s) per core:  2
Core(s) per socket:  18
Socket(s):           2
NUMA node(s):        2
Vendor ID:           GenuineIntel
CPU family:          6
Model:               85
Model name:          Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz
Stepping:            7
CPU MHz:             3563.885
CPU max MHz:         3900.0000
CPU min MHz:         1000.0000
BogoMIPS:            5200.00
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            1024K
L3 cache:            25344K
NUMA node0 CPU(s):   0-17,36-53
NUMA node1 CPU(s):   18-35,54-71
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke avx512_vnni md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.7.1
[pip3] flake8==6.0.0
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.26.4
[pip3] numpydoc==1.5.0
[pip3] perlin-numpy==0.0.0
[pip3] pytorch-lightning==2.2.3
[pip3] torch==2.0.1
[pip3] torch-cluster==1.6.3
[pip3] torch_geometric==2.4.0
[pip3] torch-scatter==2.1.2
[pip3] torch-sparse==0.6.18
[pip3] torchaudio==2.0.2
[pip3] torchdata==0.7.1
[pip3] torchmetrics==1.3.2
[pip3] torchvf==0.1.3
[pip3] torchvision==0.15.2
[pip3] triton==2.0.0
[conda] blas                      1.0                         mkl  
[conda] efficientnet-pytorch      0.7.1                    pypi_0    pypi
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0           py310h7f8727e_0  
[conda] mkl_fft                   1.3.1           py310hd6ae3a3_0  
[conda] mkl_random                1.2.2           py310h00e6091_0  
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] numpydoc                  1.5.0           py310h06a4308_0  
[conda] perlin-numpy              0.0.0                    pypi_0    pypi
[conda] pytorch-lightning         2.2.3                    pypi_0    pypi
[conda] torch                     2.0.1                    pypi_0    pypi
[conda] torch-cluster             1.6.3                    pypi_0    pypi
[conda] torch-geometric           2.4.0                    pypi_0    pypi
[conda] torch-scatter             2.1.2                    pypi_0    pypi
[conda] torch-sparse              0.6.18                   pypi_0    pypi
[conda] torchaudio                2.0.2                    pypi_0    pypi
[conda] torchdata                 0.7.1                    pypi_0    pypi
[conda] torchmetrics              1.3.2                    pypi_0    pypi
[conda] torchvf                   0.1.3                    pypi_0    pypi
[conda] torchvision               0.15.2                   pypi_0    pypi
[conda] triton                    2.0.0                    pypi_0    pypi

**Comment by hengck23** [comment]
It is my mistake. I am very sorry.
The leak is caused by somewhere else.
